### PR TITLE
UMAPINFO: bring par times fixes in line with PrBoom+

### PR DESCRIPTION
**Pull request by fabiangreffrath**
https://github.com/coelckers/prboom-plus/commit/ee33927e3688668b2010cd82ecc1d352cff18391 https://github.com/coelckers/prboom-plus/commit/d97cb0e78558b83e8c33d9a1d0129570f5b7563e

### CODE REVIEW

File: src/g_game.c [hunk 1]
```diff
@@ -1382,10 +1367,13 @@ static void G_DoCompleted(void)
...
-  if ( gamemode == commercial )
-    wminfo.partime = TICRATE*cpars[gamemap-1];
-  else
-    wminfo.partime = TICRATE*pars[gameepisode][gamemap];
+  if (!(gamemapinfo && gamemapinfo->partime))
+  {
+    if ( gamemode == commercial )
+      wminfo.partime = TICRATE*cpars[gamemap-1];
+    else
+      wminfo.partime = TICRATE*pars[gameepisode][gamemap];
```
Comment (fabiangreffrath) [inline src/g_game.c:1375]: What will happen in Ultimate Doom's Episode 4 here?!

**Comment by rfomin** [comment]
I thought we'd fix PrBoom+ instead to bring PrBoom+ in line with its previous versions and with DSDA-Doom.

**Comment by fabiangreffrath** [comment]
> I thought we'd fix PrBoom+ instead to bring PrBoom+ in line with its previous versions and with DSDA-Doom.

PrBoom+ is considered to be the fix:
https://www.doomworld.com/forum/topic/112333-this-is-woof-1030-sep-23-2022-updated-winmbf/?do=findComment&comment=2549546

**Comment by kraflab** [comment]
But woof is the fix that doesn't break demo sync right?

**Comment by fabiangreffrath** [comment]
Ah, I've just seen that you also apply `partime` only in the `if next[0]` case in DSDA:

https://github.com/kraflab/dsda-doom/blob/019e8d23674db0077814ab1d47660df6b1ab218f/prboom2/src/dsda/mapinfo/u.c#L425

Apparently you are only missing the check to actually print them on the intermission screen:

https://github.com/kraflab/dsda-doom/blob/019e8d23674db0077814ab1d47660df6b1ab218f/prboom2/src/wi_stuff.c#L952

File: src/g_game.c (hunk 1 above)
Comment (fabiangreffrath) [reply src/g_game.c:1375]: Wow, this overflows straight into the `cpars[]` array for Episode 4! :fearful:

**Pull request closed without merging**